synapse/util/retryutils.py

```diff
@@ -150,10 +160,10 @@ def __init__(
         self.backoff_on_404 = backoff_on_404
         self.backoff_on_failure = backoff_on_failure
 
-    def __enter__(self):
+    def __enter__(self) -> None:
         pass
 
-    def __exit__(self, exc_type, exc_val, exc_tb):
+    def __exit__(self, exc_type, exc_val, exc_tb: TracebackType) -> None:
```

Comment:
Aaaaaand the other args?

Reply:
I didn't know how to type these, but actually I think I can do it with `cast`

Reply:
The types are `exc_type: Optional[Type[BaseException]], exc_val: Optional[BaseException],  exc_tb: Optional[TracebackType]` by the sounds of things:  https://github.com/python/typeshed/blob/master/stdlib/typing.pyi#L426